1-Draft/RFCXXXX-Obsoletion-and-Deprecation.md

```diff
@@ -0,0 +1,511 @@
+
+PS> Get-ObsoleteItem -ItemType Command
+
+   ItemType: Command
+
+FullyQualifiedWarningId                  ObsoleteMembers           Message
+------------------                       ---------------           -------
+Microsoft.PowerShell.Utility\Format-Hex… Raw                       Raw parameter is obsolete.
+Microsoft.PowerShell.Utility\Send-MailM… none                      This cmdlet does not guarantee secure connections to SMTP servers. While there is …
+
+```
+
+### `Set-ObsoleteBehavior`
+
+  This cmdlet enables an item which specific override behavior for a specific WarningId.
+  If a WarningId setting has already been set, it will be overridden with the new action value.
+
+  ```powershell
+  PS> Set-ObsoleteBehavior -WarningId Microsoft.PowerShell.Utility\7.0.0.0\Format-Hex:Raw -Action SilentlyContinue
+  PS> Set-ObsoleteBehavior -WarningId /Users/james/bin/invoke-obsolete.ps1:foo -Action Continue
+  ```
+
+### `Get-ObsoleteBehavior`  
+
+This command returns all the currently configured warning behavior.
+By default it will return configured settings for those items which are currently available in the session.
+The `-All` parameter will search through all modules (loaded and unloaded) and script returning the current behavior for all discovered obsolete items.
+
+```powershell
+PS> Get-ObsoleteBehavior
+
+FullyQualifiedWarningId                                                     Action
+-----------------------                                                     ------
+Microsoft.PowerShell.Utility\Format-Hex:Raw                                 SilentlyContinue
+Microsoft.PowerShell.Utility\Send-MailMessage                               Error
+System.Management.Automation.Runspaces.InitialSessionState:ImportPSSnapIn   NotFound
+
+PS> Get-ObsoleteBehavior -All
+
+FullyQualifiedWarningId                                                     Action
+-----------------------                                                     ------
+Microsoft.PowerShell.Utility\Format-Hex:Raw                                 SilentlyContinue
+/Users/james/bin/invoke-obsolete.ps1:foo                                    Continue
+Microsoft.PowerShell.Utility\Send-MailMessage                               Error
+System.Management.Automation.Runspaces.InitialSessionState:ImportPSSnapIn   NotFound
+```
+
+## Implementation Considerations
+
+- WarningId does not generally ever have a value in the current WarningRecord.
+This would need to be used consistently in order for suppression as defined above to work.
+
+- With regard to interactive sessions, PSReadLine does many pre-fetches when the user types the `[...` while referencing types.
+Currently, PSReadLine behavior puts types into the type cache even if they are not actually used.
+This will likely have implications for reporting obsolete types and members in an interactive session.
```

Comment:
The pre-fetch happens if a type is resolved at parsing time, or if tab completion or similar intellisense operations take place. All type resolution should only happen in PowerShell engine, not PSReadLine, so it won't affect anything if the engine decides to not resolve (cannot find) a type or member.